### PR TITLE
Never Give Up: Learning Directed Exploration Strategies

### PR DESCRIPTION
**Pull request by AntoineTheb**
ngu

TODO: Add poster https://iclr.cc/virtual_2020/poster_Sye57xStvB.html

### CODE REVIEW

**Comment by AntoineTheb** [comment]
Le hook est brisé, il permet pas de mettre une image comme contenu de lien.

**Comment by AntoineTheb** [comment]
@jodoin @ppoulin91 @jhlegarreta @lemairecarl @nathanpainchaud @MarcoArmenta please review even though the build has failed; one of the hooks is broken.

File: article/_posts/2020-05-28-NGU.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+---
+layout: review
+title: "Never Give Up: Learning Directed Exploration Strategies"
+tags:
```
Comment (jhlegarreta) [inline article/_posts/2020-05-28-NGU.md:4]: Tags are missing.

File: article/_posts/2020-05-28-NGU.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+A learning agent often needs to finetune a tradeoff between exploitation (acting on what it knows will work) and exploration (behaving in a way that is not guaranteed to be good) so as to learn a policy that achieves good reward without being stuck in a local optimum.
+
+The first ideas for exploration were to have an $${\epsilon}-greedy$$ policy (taking a random action with a probability of $$1-\epsilon$$) or to do Boltzmann exploration (exploring w.r.t the Q function of the state-action pair with a temperature parameter to decrease exploration over time). However, these methods are very inefficient in the context of larger non-tabular state-spaces. 
+
+Recent approaches have proposed to equip agents with an intrinsic reward promoting exploration, making them focus on exploring new, unseen states. As the agent explores its environment, the intrinsic motivation wears off and the reward becomes extrinsic. However, once a state has been visited, the agent is no longer encouraged to visit it again and therefore might not explore subsequent states that might have been more profitable. Finetuning the "curiosity" only seems to delay the problem.
```
Comment (jhlegarreta) [inline article/_posts/2020-05-28-NGU.md:30]: The curiosity is the intrinsic reward promoting exploration, right? Maybe you can add the former between brackets when you first mention the later in the paragraph.

File: article/_posts/2020-05-28-NGU.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+To alleviate the problem, the authors $$(i)$$ use a recurrent agent based on R2D2[^2] and $$(ii)$$ also feed the intrinsic reward, amongst other things, as an input to the network.
+
+![](/article/images/NGU/fig17.jpeg)
+
+To train a variety of exploring and exploiting policies, the authors use a universal value function approximator (UVFA)[^3] $$Q(x, a, \beta)$$ to approximate the optimal value function w.r.t the reward of the form $$r_t = r_{t}^{e} + {\beta}{_i}{r_{t}^{i}}$$ where $$\beta{_i}$$ is apart of a family of $$N$$ values $$\{\beta{_i}\}_{i=0}^{N-1}$$ including $$\beta_{0} = 0$$. Having a large N means having a greater variety in behavior and therefore more efficient training.
+
+## Experiments
+
+The authors tested on the Atari suited as well as their own new environment: <i>Random Disco Maze</i>.
```
Comment (jhlegarreta) [inline article/_posts/2020-05-28-NGU.md:62]: _Atari suite_? _Atari suites_?

File: article/_posts/2020-05-28-NGU.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+The authors tested on the Atari suited as well as their own new environment: <i>Random Disco Maze</i>. 
+
+![](/article/images/NGU/fig2.jpeg)
+
+The agent, in green, must navigate the maze, in black. If the agent takes an action that would make it go outside the maze, the experiment stops. At each time step, the colors of the background changes, and there is no extrinsic reward.
+
+# Results
+
+Fig 2 shows the exploration does by the agent on <i>Random Disco Maze</i>, and [this video](https://www.youtube.com/watch?v=9HTY4ruPrHw&feature=youtu.be) shows the agent in action.
```
Comment (jhlegarreta) [inline article/_posts/2020-05-28-NGU.md:70]: _done by the agent_?

File: article/_posts/2020-05-28-NGU.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+To train a variety of exploring and exploiting policies, the authors use a universal value function approximator (UVFA)[^3] $$Q(x, a, \beta)$$ to approximate the optimal value function w.r.t the reward of the form $$r_t = r_{t}^{e} + {\beta}{_i}{r_{t}^{i}}$$ where $$\beta{_i}$$ is apart of a family of $$N$$ values $$\{\beta{_i}\}_{i=0}^{N-1}$$ including $$\beta_{0} = 0$$. Having a large N means having a greater variety in behavior and therefore more efficient training.
+
+## Experiments
+
+The authors tested on the Atari suited as well as their own new environment: <i>Random Disco Maze</i>. 
+
+![](/article/images/NGU/fig2.jpeg)
+
+The agent, in green, must navigate the maze, in black. If the agent takes an action that would make it go outside the maze, the experiment stops. At each time step, the colors of the background changes, and there is no extrinsic reward.
```
Comment (jhlegarreta) [inline article/_posts/2020-05-28-NGU.md:66]: _the colors of the background change_
or
_the color of the background changes_

**Comment by lemairecarl** [comment]
Great. I love the Random Distillation Idea.

**Pull request merged**